external-crates/move/crates/move-vm-runtime/src/interpreter/mod.rs

```diff
@@ -0,0 +1,59 @@
+use crate::{
+    interpreter::state::{CallFrame, MachineState},
+    loader::{arena::ArenaPointer, ast::Function, Loader},
+};
+use move_binary_format::errors::*;
+use move_vm_profiler::{profile_close_frame, profile_open_frame};
+use move_vm_types::{
+    data_store::DataStore, gas::GasMeter, loaded_data::runtime_types::Type, values::Value,
+};
+
+use crate::native_extensions::NativeContextExtensions;
+
+mod eval;
+pub(crate) mod state;
+
+/// Entrypoint into the interpreter. All external calls need to be routed through this
+/// function.
+pub(crate) fn run(
+    function: ArenaPointer<Function>,
```

Comment:
Lots to think about here for me. At some point in the far future, it would be very interesting to basically call `step`, or at least "run" here to invoke a function, with a given state, instead of having the custom execution engine we have today for PTBs